tasks/node/browserstack.js

```diff
@@ -0,0 +1,218 @@
+'use strict';
+
+const denodeify = require('denodeify');
+
+if (!process.env.BROWSERSTACK_USERNAME || !process.env.BROWSERSTACK_ACCESS_KEY) {
+	throw new Error('BROWSERSTACK_USERNAME and BROWSERSTACK_ACCESS_KEY must be set in the environment to run tests on BrowserStack.  For more information about how to set this up or for alternative methods of testing, see https://polyfill.io/v2/docs/contributing/testing');
+}
+
+module.exports = {
+	creds: {
+		username: process.env.BROWSERSTACK_USERNAME,
+		key: process.env.BROWSERSTACK_ACCESS_KEY
+	},
+	tunnel: function () {
```

Comment:
Each provider has a different way to configure their tunnel, in order to simplify the `remotetest.js` code I've decided to have a standard interface for opening and closing tunnels. It is a basic object which has two methods which return Promises: `openTunnel`, `closeTunnel`.